uvm/servlets/admin/config/network/Interface.js

```diff
@@ -37,7 +37,25 @@ Ext.define('Ung.config.network.Interface', {
                 allowOnlyWhitespace: false,
                 regex: /^[^!#$%^&]+$/,
                 regexText: 'This field can have alphanumerics or special characters other than ! # $ % ^ &'.t(),
-                bind: '{intf.name}'
+                bind: '{intf.name}',
+                validator: function(value) {
+                    var store = this.up('tabpanel').getViewModel().getStore('interfaces');
+                    var title = this.up('window').title;
+                    var currentInterfaceName = this.up('window').getViewModel().get('intf.name');
+                
+                    // Check if a record with the same name exists in the store
+                    var isNameUnique = store.findBy(function(record) {
+                        return record.get('name') === value;
+                    }) === -1;
+                
+                    // If the title indicates it's adding a VLAN interface, simply check for uniqueness
+                    if (title === 'Add VLAN Interface') {
```

Comment:
Definitely don't compare text strings for windows.  I think your "else" clause check is sufficient.  Customers will see any type of interface (physical or VLAN) as an "interface" and don't need the differentiation.

Reply:
Thanks @cblaise-untangle for confirming on users view and for your suggestion. Implemented and tested the same.